test/hotspot/jtreg/compiler/c2/irTests/TestFPConversion.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package compiler.c2.irTests;
+
+import compiler.lib.ir_framework.*;
+import jdk.test.lib.Asserts;
+
+/*
+ * @test
+ * @bug 8345219
+ * @summary Test that code generation for FP conversion works as intended
+ * @library /test/lib /
+ * @requires os.arch != "x86" & os.arch != "i386"
```

Comment:
It would have been preferrable to add this to the IR rule, so the test still runs elsewhere.
What about `aarch64`?

Reply:
For all practical purposes, the test will run everywhere, really. It would be skipped only on `IA32`, which is going away. This `@requires` serves as the concession to `IA32` that is still in the tree.

Reply:
Ah, I read the requires wrong. All good.